Landry333/Big-Owl
# Us 71 notifications

### Related Issues
#71 

### Description
I refactored the NotificationListenerManager (originally AuthFailureNotificationListener).
I added notification functions in SignUpViewModel.
I added a few functions in smsInvitationActivity.
I modified the notification so it return a string not an enum.
I made a "generic" notification channel and builder in NotificationListenerManager so it can adapt to most notification if not all.


### How to check PR
To test:
1. Login as a user.
2. add user by phone number (the user must not exist in the db)
3. Once it is done, log out and registered a new user with the SAME PHONE NUMBER that it was sent previously.
4. Once registered, a notification will pop up for all user that send an invitation.
5. A notification should pop up 

if there is any questions, please comment. THANK YOU

PS: I cannot request @refatK because he help me with this feature. We made a lot of change.
